umbral/openssl.py

```diff
@@ -62,7 +63,10 @@ def _bn_is_on_curve(check_bn, curve_nid: int):
     """
     ec_order = _get_ec_order_by_curve_nid(curve_nid)
 
-    check_sign = backend._lib.BN_cmp(check_bn, backend._int_to_bn(0))
+    zero = backend._int_to_bn(0)
+    backend._ffi.gc(zero, backend._lib.BN_clear_free)
```

Comment:
Do we need to call the `gc` before or after the next line which does the comparison?

Reply:
You can call it immediately after instantiation, so before. It can also be done after, but that seems like a risky/problematic pattern.